cmd/skaffold/app/cmd/config/set.go

```diff
@@ -46,12 +46,27 @@ func NewCmdSet(out io.Writer) *cobra.Command {
 	return cmd
 }
 
-func setConfigValue(name string, value interface{}) error {
+func setConfigValue(name string, value string) error {
```

Comment:
The reason I originally had it as an `interface{}` was mostly for clarity that even if the provided value can be represented as a string, it might semantically be something different, i.e. a number, or even a nested struct. Is there any reason we need to make it a string here?

Reply:
Right, that's a good reason.

In my new implementation, the type actually needs to be a string, because the `getNewFieldValue` function builds an auxiliary yaml from the input value.

Clearly, your previous version for `setConfigValue` is more generic. On the other hand, it doesn't handle type conversion of the given value into the desired target type. To me, adapting `setConfigValue` seemed to be the right place for the type conversion. But maybe the type conversion should be done in the caller of `setConfigValue`, then the original code would work just fine. What do you think?